### PR TITLE
Early out and report error if langauge files are missing

### PR DESCRIPTION
**Pull request by janisozaur**
This is a common problem if you don't set up your paths properly on
Linux, so be nice and try to provide helpful message why we failed to
load instead of crashing on user a moment later.

### CODE REVIEW

File: src/openrct2.c [hunk 1]
```diff
@@ -225,7 +225,11 @@ bool openrct2_initialise()
 		audio_init();
 		audio_get_devices();
 	}
-	language_open(gConfigGeneral.language);
+	if (!language_open(gConfigGeneral.language))
+	{
+		log_error("Failed to open language, exiting.");
+		exit(1);
```
Comment (duncanspumpkin) [inline src/openrct2.c]: elsewhere in this function we return false on a fatal error. Also this should be `log_fatal` for this one

File: src/windows/options.c [hunk 2]
```diff
@@ -1011,10 +1011,15 @@ static void window_options_dropdown(rct_window *w, int widgetIndex, int dropdown
 			break;
 		case WIDX_LANGUAGE_DROPDOWN:
 			if (dropdownIndex != gCurrentLanguage - 1) {
-				language_open(dropdownIndex + 1);
-				gConfigGeneral.language = dropdownIndex + 1;
-				config_save_default();
-				gfx_invalidate_screen();
+				if (!language_open(dropdownIndex + 1))
+				{
+					log_error("Failed to change language.");
```
Comment (IntelOrca) [inline src/windows/options.c]: We can use an in-game error here using window_error_open(...)

**Comment by Gymnasiast** [comment]
That is the one string that _shouldn't_ be in a language file, as it will only display the error when it _can't_ load the correct language file. It should be hardcoded.

**Comment by IntelOrca** [comment]
Sad but true, and in this case we can't just choose a language file at random to get this string from.

**Comment by janisozaur** [comment]
But if user got to this point, that means he had a language file in the first place, as otherwise `openrct2_initialise()` wouldn't let him through.

I guess the correct thing would be to do `if(language_open(prev_lang))` then display warning window in-game, and only fail otherwise, as that means that language file was removed and we will probably crash soon anyway.

**Comment by IntelOrca** [comment]
oh sorry, I thought we were talking about initialise

**Comment by janisozaur** [comment]
I couldn't find a way to properly do `window_error_open` in case of language load failure, so I dumbed it down to the basics and just report error to log. `language_open` sets a fallback to english-uk, so this will have to do. It's rather unlikely that files start missing during gameplay, so I don't expect too many issues here, especially if next launch will fail on exactly this problem with `FATAL` in log.

**Comment by IntelOrca** [comment]
What are you having difficulty with when trying to use window_error_open?

**Comment by janisozaur** [comment]
it always shows `(undefined string)`

**Comment by IntelOrca** [comment]
What are you passing as the arguments?

**Comment by janisozaur** [comment]
I realized that I was using wrong `data` directory, which did not include the update language file. It should now work.

**Pull request merged**